packages/xo-web/src/xo-app/dashboard/overview/index.js

```diff
@@ -105,84 +152,98 @@ class PatchesCard extends Component {
       return metrics
     })
   )
-  const getNumberOfAlarmMessages = createCounter(
-    createGetObjectsOfType('message'),
-    [message => message.name === 'ALARM']
-  )
-  const getNumberOfHosts = createCounter(getHosts)
-  const getNumberOfPools = createCounter(createGetObjectsOfType('pool'))
-  const getNumberOfTasks = createCounter(
-    createGetObjectsOfType('task').filter([task => task.status === 'pending'])
+
+  _getSrMetrics = createCollectionWrapper(
+    createSelector(this._getSrs, srs => {
+      const metrics = {
+        srTotal: 0,
+        srUsage: 0,
+      }
+      forEach(srs, sr => {
+        metrics.srUsage += sr.physical_usage
+        metrics.srTotal += sr.size
+      })
+      return metrics
+    })
   )
-  const getNumberOfVms = createCounter(getVms)
 
-  return {
-    hostMetrics: getHostMetrics,
-    hosts: getHosts,
-    nAlarmMessages: getNumberOfAlarmMessages,
-    nHosts: getNumberOfHosts,
-    nPools: getNumberOfPools,
-    nTasks: getNumberOfTasks,
-    nVms: getNumberOfVms,
-    srMetrics: getSrMetrics,
-    topWritableSrs: createTop(
-      writableSrs,
-      [sr => sr.physical_usage / sr.size],
-      5
-    ),
-    vmMetrics: getVmMetrics,
-  }
-})
-@injectIntl
-class DefaultCard extends Component {
-  componentWillMount () {
-    this.componentWillUnmount = subscribeUsers(users => {
-      this.setState({ users })
+  _getTopSrs = createTop(this._getSrs, [sr => sr.physical_usage / sr.size], 5)
+
+  _onChange = hosts => {
+    this.setState({
+      hosts: compact(hosts),
```

Comment:
Why do you need to `compact` it? Maybe selecting new pools should change `this.state.hosts`?